swimlane/core/resources/revision_base.py

```diff
@@ -0,0 +1,45 @@
+import pendulum
+
+from swimlane.core.resources.base import APIResource
+from swimlane.core.resources.usergroup import UserGroup
+
+
+class RevisionBase(APIResource):
+    """
+    The base class representing a single revision returned from a History lookup.
+
+    Attributes:
+        Attributes:
+        modified_date: The date this revision was created.
+        revision_number: The revision number of this revision.
+        status: Indicates whether this revision is the current revision or a historical revision.
+        user: The user that saved this revision.
+    """
+
+    def __init__(self, swimlane, raw):
+        super(RevisionBase, self).__init__(swimlane, raw)
+
+        self.modified_date = pendulum.parse(self._raw['modifiedDate'])
+        self.revision_number = self._raw['revisionNumber']
+        self.status = self._raw['status']
+
+        # UserGroupSelection, can't set as User without additional lookup
+        self.user = UserGroup(self._swimlane, self._raw['userId'])
+
+        self._raw_version = self._raw['version']
+        self._version = None
```

Comment:
The way I am doing this property seems wrong to me. I tried to do `__version` but then could not access from the child classes. Is there a better way to do this?

Reply:
here is why you cant access `self.__version` from outside the class https://stackoverflow.com/questions/797771/python-protected-attributes
TLDR; the var is renamed to `_Class__variable` 

thats pythons way of doing private vars, if you need to access outside the class at all the single underscore is your best bet

Reply:
https://stackoverflow.com/questions/1301346/what-is-the-meaning-of-a-single-and-a-double-underscore-before-an-object-name/12629901#12629901

Reply:
Thanks for the links, I think I understand this better now. It looks to me that since I want to access this in the child classes, single underscore like I have now is the cleanest way to go? Or was there something else you had in mind?